components/api_request_helper/api_request_helper.cc

```diff
@@ -93,8 +135,19 @@ void APIRequestHelper::OnResponse(
     }
   }
   url_loaders_.erase(iter);
-  std::move(callback).Run(response_code, response_body ? *response_body : "",
-                          headers);
+  if (!response_body) {
+    std::move(callback).Run(response_code, "", headers);
+    return;
+  }
+  std::string raw_body_to_patch = *response_body;
+  if (conversion_callback)
+    std::move(conversion_callback).Run(&raw_body_to_patch);
+
+  data_decoder::JsonSanitizer::Sanitize(
+      std::move(raw_body_to_patch),
```

Comment:
hmm, I don't think it's safe to modify it in callback and access it here, is it?

Reply:
You should probably use PostTaskAndReplyWithResult or so instead.

Reply:
I double checked again, seems okay to use like this.
https://source.chromium.org/chromium/chromium/src/+/main:base/callback.h;l=134-142?q=callback%20Run&ss=chromium

Example of chromium usage that's similar https://source.chromium.org/chromium/chromium/src/+/main:extensions/browser/file_reader.cc;l=50-51